### PR TITLE
Day9-Bai1-Feed_Activity

### PR DESCRIPTION
**Pull request by at-datbui**
#### Description (Required)

- [x] Create Feed Activity
- [x] get image from drawable
#### Reference (Optional)

### CODE REVIEW

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package vn.asiantech.internship.feed;
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 15-06-2017.
+ */
+
+class Feed {
+    private String Name;
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: ERROR: Name 'Name' must match pattern '^[a-z]([_a-zA-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    private int[] Image;
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: ERROR: Name 'Image' must match pattern '^[a-z]([_a-zA-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    private String Title;
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: ERROR: Name 'Title' must match pattern '^[a-z]([_a-zA-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/feed/ViewPagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+class ViewPagerAdapter extends PagerAdapter {
+    private Context mContext;
+    private int[] mSampleImages;
+
+    ViewPagerAdapter(Context context, int[] sampleImages) {
+        mSampleImages = sampleImages;
+        mContext = context;
+    }
+
+    void setImage(int[] SampleImages) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/ViewPagerAdapter.java]: ERROR: Name 'SampleImages' must match pattern '^[a-z]([_a-zA-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/feed/FeedFragment.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_feed, container, false);
+
+        List<Feed> feedList = new ArrayList<>();
+        int[] sampleImages = {R.drawable.image_1, R.drawable.image_2, R.drawable.image_3, R.drawable.image_4, R.drawable.image_5};
+        feedList.add(new Feed("dasdas", sampleImages, "1"));
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/FeedFragment.java]: WARNING: [AvoidDuplicateLiterals] The String literal "dasdas" appears 9 times in this file; the first occurrence is on line 35
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -7,5 +7,17 @@
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
     </style>
-
+    <style name="txt">
```
Comment (at-dev-support-01) [inline app/src/main/res/values/styles.xml]: WARNING: [UnusedResources] The resource `R.style.txt` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/vn/asiantech/internship/feed/ActivityFeed.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import android.support.v4.app.FragmentTransaction;
+import android.support.v7.app.AppCompatActivity;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 15-06-2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/ActivityFeed.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
+package vn.asiantech.internship.feed;
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 15-06-2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    int[] getImage() {
+        return image;
+    }
+
+    String getTitle() {
+        return title;
+    }
+
+    Feed(String name, int[] image, String title) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: move this method to` line 13`

File: app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 15-06-2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
+package vn.asiantech.internship.feed;
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 15-06-2017.
+ */
+class Feed {
+    private String name;
+    private int[] image;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: change variable name to `images`

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    private String title;
+
+    Feed(String name, int[] image, String title) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: change `image` to `images`

File: app/src/main/res/layout/item_image_feed.xml [hunk 13]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_image_feed.xml]: this ís `RelativeLayout`, remove `android:orientation="vertical"`

File: app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    private static final String KEY_NAME = "name";
+    private static final String KEY_TITLE = "title";
+
+    public DataBaseHandler(Context context) {
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        String CREATE_CONTACTS_TABLE = "CREATE TABLE" + TABLE_CONTACTS + "("
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java]: ERROR: Name 'CREATE_CONTACTS_TABLE' must match pattern '^[a-z]([_a-zA-Z0-9]+)*$'.

**Comment by at-vinhhuynh** [comment]
@at-datbui 
Please fix CI first

File: app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+
+    DataBaseHandler(Context context) throws IOException {
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+        this.mContext = context;
+        mPath = mContext.getFilesDir().getPath() + File.separatorChar;
+        boolean dbexist = checkdatabase();
+        if (dbexist) {
+            opendatabase();
+        } else {
+            System.out.println("Database doesn't exist");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java]: ERROR: [SystemPrintln] System.out.println is used
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#SystemPrintln

File: app/src/main/java/vn/asiantech/internship/feed/FeedFragment.java [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+        View view = inflater.inflate(R.layout.fragment_feed, container, false);
+
+        DataBaseHandler dataBaseHandler;
+        try {
+            dataBaseHandler = new DataBaseHandler(getContext());
+            dataBaseHandler.opendatabase();
+            mFeedList = dataBaseHandler.getFeeds();
+            dataBaseHandler.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/FeedFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+        if (mDb != null) {
+            mDb.close();
+        }
+        super.close();
+    }
+
+    List<Feed> getFeeds() {
+        List<Feed> feeds = new ArrayList<>();
+        opendatabase();
+        Cursor cursor = mDb.rawQuery("SELECT * FROM " + TABLE_NAME, null);
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java]: WARNING: [Recycle] This `Cursor` should be freed up after use with `#close()`
 Many resources, such as TypedArrays, VelocityTrackers, etc., should be recycled (with a `recycle()` call) after use. This lint check looks for missing `recycle()` calls.

File: app/src/main/res/values/strings.xml [hunk 18]
```diff
@@ -1,11 +1,11 @@
...
-
+    <string name="app_name">MyProject</string>
     <string name="get_text">Get Text</string>
     <string name="pass_data">Pass Data</string>
 
-    <string name="app_name">MyProject</string>
     <string name="btn_friend">Friend</string>
     <string name="btn_add">Add</string>
     <string name="tv_value">asdsadasdsadsad</string>
+    <string name="feedlist.name">dasdas</string>
```
Comment (at-dev-support-01) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.feedlist_name` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static android.content.ContentValues.TAG;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java]: remove `import static`

File: app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java [hunk 20]
```diff
@@ -0,0 +1,119 @@
...
+        mPath = mContext.getFilesDir().getPath() + File.separatorChar;
+        boolean dbexist = checkdatabase();
+        if (dbexist) {
+            opendatabase();
+        } else {
+            createdatabase();
+        }
+    }
+
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/DataBaseHandler.java]: remove a break line

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 15-06-2017.
+ */
+class Feed {
+    private int id;
+    private String name;
+    private String title;
+    private String[] strimages;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: variable name `strImages`

File: app/src/main/java/vn/asiantech/internship/feed/Feed.java [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+
+    Feed(String name, String strimages, String title) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/Feed.java]: strimages to `strImages`

File: app/src/main/res/layout/item_image_feed.xml [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+    xmlns:card_view="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/cardView"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_margin="5dp"
+    card_view:cardCornerRadius="4dp"
+    card_view:cardElevation="2dp"
+    card_view:cardUseCompatPadding="true">
+
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_image_feed.xml]: remove break line

**Comment by at-tamnguyen2** [comment]
Thanks @at-datbui 
LGTM

**Comment by at-tienhoang** [comment]
LGTM! thank you
LGTM

**Pull request merged**